--- a/src/Carbon/CarbonPeriod.php
+++ b/src/Carbon/CarbonPeriod.php
@@ -38,6 +38,7 @@
 use DateTimeInterface;
 use DateTimeZone;
 use Generator;
+use Iterator;
 use InvalidArgumentException;
 use JsonSerializable;
 use ReflectionException;
@@ -326,7 +327,7 @@
      */
     protected DateTimeZone|string|int|null $timezoneSetting = null;
 
-    public function getIterator(): Generator
+    public function getIterator(): Iterator
     {
         $this->rewind();
 
